addons/l10n_ec/models/account_move.py

```diff
@@ -0,0 +1,153 @@
+        return None
+
+    @api.onchange('product_id')
+    def _onchange_product_id(self):
+        for line in self:
+            if not line.product_id or line.display_type in ('line_section', 'line_note'):
+                continue
+
+            line.name = line._get_computed_name()
+            line.account_id = line._get_computed_account()
+            line.tax_ids = line._get_computed_taxes()
+            line.witholding_tax_ids = line._get_computed_wth()
+            line.product_uom_id = line._get_computed_uom()
+            line.price_unit = line._get_computed_price_unit()
+
+            # price_unit and taxes may need to be adapted following Fiscal Position
+            line._set_price_and_tax_after_fpos()
+
+            # Convert the unit price to the invoice's currency.
+            company = line.move_id.company_id
+            line.price_unit = company.currency_id._convert(line.price_unit, line.move_id.currency_id, company, line.move_id.date, round=False)
+
+    def _set_price_and_tax_after_fpos(self):
+        self.ensure_one()
+        super(AccountMoveLine,self)._set_price_and_tax_after_fpos()
+        if self.tax_ids and self.move_id.fiscal_position_id and self.move_id.fiscal_position_id.tax_ids:
+            self.witholding_tax_ids = self.move_id.fiscal_position_id.map_tax(
+                self.witholding_tax_ids._origin,
+                partner=self.move_id.partner_id)
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    amount_by_group_wth = fields.Binary(string="Witholding amount by group",
+        compute='_compute_invoice_witholding_by_group',
+        help='Edit Tax amounts if you encounter rounding issues.')
+
+    l10n_ec_sri_payment_id = fields.Many2one('l10n.ec.sri.payment', _("Payment Method (SRI)"))
+    l10n_ec_access_key = fields.Char(_("Authorization"), copy=False)
+    l10n_ec_auth_type = fields.Selection(related="l10n_latam_document_type_id.l10n_ec_authorization")
+    l10n_ec_is_electronic = fields.Boolean(default=False, compute="_l10n_ec_is_electronic")
+
+    def _get_l10n_latam_documents_domain(self):
+        #Filter document types according to ecuadorian move_type
+        domain = super(AccountMove, self)._get_l10n_latam_documents_domain()
+        if self.country_code == 'EC':
+            if self.move_type in ['out_invoice']:
+                domain.extend([('l10n_ec_type', '=', 'out_invoice')])
+            if self.move_type in ['out_refund']:
+                domain.extend([('l10n_ec_type', '=', 'out_refund')])
+            if self.move_type in ['in_invoice']:
+                domain.extend([('l10n_ec_type', '=', 'in_invoice')])
+            if self.move_type in ['in_refund']:
+                domain.extend([('l10n_ec_type', '=', 'in_refund')])
+        return domain
+
+    @api.depends('journal_id', 'partner_id')
+    def _l10n_ec_is_electronic(self):
+        self.ensure_one()
+        self.l10n_ec_is_electronic = len(self.journal_id.edi_format_ids) > 0
```

Comment:
This is a weird condition, because any other format could activate this.  Normally the logic for handling the electronic part is done through the methods of the edi format itself.